src/random.cpp

```diff
@@ -248,7 +248,9 @@ static void Strengthen(const unsigned char (&seed)[32], int microseconds, CSHA51
     memory_cleanse(buffer, sizeof(buffer));
 }
 
-#ifndef WIN32
+#if !defined(WIN32) && \
+  !(defined(HAVE_GETENTROPY) && defined(__OpenBSD__)) && \
```

Comment:
Concept ACK, but I think there's some redundancy in this experession: at the least, WIN32 will never be set if HAVE_GETENTROPY or HAVE_SYSCTL_ARND.